Plugins~/Src/MeshSyncClientBlender/python/unity_mesh_sync_baking.py

```diff
@@ -0,0 +1,1202 @@
+
+        # Make sure meshsync is finished and ready:
+        from .unity_mesh_sync_common import msb_apply_scene_settings, msb_context
+
+        # if not msb_context.is_setup:
+        msb_context.flushPendingList()
+        msb_apply_scene_settings()
+        msb_context.setup(context)
+
+        activeObject = context.object
+        selectedObjects = context.selected_objects
+
+        self.objectsProcessedForBaking = []
+        bakeSettings = context.scene.meshsync_bake_settings
+        bakeSettings.bake_progress = 0
+
+        bakeSelection = bakeSettings.bake_selection
+
+        if bakeSelection == 'ALL':
+            objectsToBake = context.scene.objects
+        elif bakeSelection == 'SELECTED':
+            if len(context.selected_objects) > 0:
+                objectsToBake = selectedObjects
+            else:
+                msb_log("No objects selected, nothing to bake!", LogLevel.ERROR)
+                return {'CANCELLED'}
+
+        self.setupRenderSettings(context)
+
+        # Make sure all collections are visible, baking won't work for objects in hidden collections:
+        hiddenCollectionsViewport = []
+        hiddenCollectionsRender = []
+
+        for col in context.scene.collection.children_recursive:
+            if col.hide_viewport:
+                col.hide_viewport = False
+                hiddenCollectionsViewport.append(col.name)
+            if col.hide_render:
+                col.hide_render = False
+                hiddenCollectionsRender.append(col.name)
+
+        self.maxBakeProgress = 0
+        self.currentBakeProgress = 0
+        for obj in objectsToBake:
+            self.preBakeObject(obj)
+
+        if bakeSettings.run_modal:
+            yield
+
+        for obj in objectsToBake:
+            for _ in self.bakeObject(obj):
+                yield
+
+        # Restore state:
+        self.restoreOriginalSettings(context)
+        bakeSettings = context.scene.meshsync_bake_settings
+        bakeSettings.bake_progress = 0
+        context.view_layer.objects.active = activeObject
+        for o in context.selected_objects:
+            o.select_set(False)
+        for o in selectedObjects:
+            o.select_set(True)
+        for hiddenCollection in hiddenCollectionsViewport:
+            for col in context.scene.collection.children_recursive:
+                if col.name == hiddenCollection:
+                    col.hide_viewport = True
+        for hiddenCollection in hiddenCollectionsRender:
+            for col in context.scene.collection.children_recursive:
+                if col.name == hiddenCollection:
+                    col.hide_render = True
+
+    def checkForUV0(self, obj, uvMapName, channel):
+        '''
+        :param obj: Object the material is on
+        :param uvMapName: Name of the UV map
+        :param channel: Input channel to process
+        :return: [True, Reason] if the given UV map name is not UV0 of the given object. Otherwise [False].
```

Comment:
The name is a bit confusing, would checkIsNotUV0 describe the function more accurately?

Reply:
I've renamed it